pkcs7/bin/server

```diff
@@ -2,26 +2,40 @@
 
 require 'json'
 require 'openssl'
-require 'sinatra'
-require 'sinatra/json'
-
-certificate = OpenSSL::X509::Certificate.new(
-  IO.read(File.expand_path('../../resource/certificate.pem', __FILE__))
-)
-store = OpenSSL::X509::Store.new
-
-post '/validate' do
-  request.body.rewind
-  parameters = JSON.parse(request.body.read, :symbolize_names => true)
-
-  signature = OpenSSL::PKCS7.new(parameters[:signature])
-  valid = signature.verify([certificate], store, parameters[:data], OpenSSL::PKCS7::NOVERIFY)
-
-  json :valid => valid,
-       :signers => (signature.signers.map do |signer|
-         {
-           :name => signer.name.to_s,
-           :signed_time => signer.signed_time.to_s
-         }
-       end)
+require 'webrick'
+
+class MyServlet < WEBrick::HTTPServlet::AbstractServlet
+  def do_POST (request, response)
+
+    certificate = OpenSSL::X509::Certificate.new(
```

Comment:
`certificate` and `store` should be defined outside of the scope of the `POST` handler so that their references can be shared between all requests.